inference-engine/src/transformations/src/transformations/low_precision/layer_transformation.cpp

```diff
@@ -0,0 +1,501 @@
+        auto newOperation = separateInStandaloneBranch(operation);
+        dequantization = NetworkHelper::getDequantization(operation, 0);
+        // TODO: It is correct to use optimizeSubtract here: uncomment following rows and fix it
+        //auto newSubtract = NetworkHelper::optimizeSubtract(dequantization.subtract);
+        //replace_node(dequantization.subtract, newSubtract);
+        NetworkHelper::removeConvertIfPossible(operation, dequantization);
+    }
+}
+
+void LayerTransformation::updateOutput(
+    TransformationContext &context,
+    std::shared_ptr<ngraph::Node> lastNode,
+    std::shared_ptr<ngraph::Node> originalNode) const {
+    const size_t outputSize = context.function->get_output_size();
+    for (size_t i = 0; i < outputSize; ++i) {
+        std::shared_ptr<ngraph::Node> result = context.function->get_output_op(i);
+        std::shared_ptr<ngraph::Node> outputNode = result->get_input_node_shared_ptr(0);
+        if (outputNode.get() == lastNode.get()) {
+            const std::string originalName = originalNode->get_friendly_name();
+            originalNode->set_friendly_name(originalName + LayerTransformation::originalLayerPostfix);
+            lastNode->set_friendly_name(originalName);
+            break;
+        }
+    }
+}
+
+void LayerTransformation::updateOutput(
+    TransformationContext& context,
+    std::shared_ptr<ngraph::Node> lastNode,
+    std::string originalName) const {
+    const size_t outputSize = context.function->get_output_size();
+    for (size_t i = 0; i < outputSize; ++i) {
+        std::shared_ptr<ngraph::Node> result = context.function->get_output_op(i);
+        std::shared_ptr<ngraph::Node> outputNode = result->get_input_node_shared_ptr(0);
+        if (outputNode.get() == lastNode.get()) {
+            lastNode->set_friendly_name(originalName);
+            break;
+        }
+    }
+}
+
+void LayerTransformation::addPattern(ngraph::pass::GraphRewrite& pass, TransformationContext& context, std::shared_ptr<Node> patternRoot) const {
+    ngraph::graph_rewrite_callback internal_callback = [this, &context](ngraph::pattern::Matcher &m) {
+        const bool result = transform(context, m);
+#ifdef LPT_DISPLAY_PRECISION
+        if (result) {
+            auto operationNode = m.get_match_root();
+            std::cout << "Operation was transformed: " <<
+                operationNode->get_type_name() << ", " <<
+                operationNode->get_friendly_name() << ", output operation precision: " <<
+                ((operationNode->get_output_size() == 1u) ? operationNode->get_output_element_type(0) : ngraph::element::Type()) <<
+                std::endl;
+        }
+#endif
+        return false;
+    };
+    // TODO: better name for matcher? required?
+    auto m = std::make_shared<ngraph::pattern::Matcher>(patternRoot, "SingleNodeMatcher");
+    pass.add_matcher(m, internal_callback, ngraph::pass::PassProperty::CHANGE_DYNAMIC_STATE);
```

Comment:
This is a deprecated way to work with matcher. Now every matcher pass must be created as MatcherPass and only then registered to GraphRewrite. See https://docs.openvinotoolkit.org/latest/ngraph_transformation.html#graph_rewrite_pass